hs_core/tests/api/rest/test_resource_types.py

```diff
@@ -0,0 +1,23 @@
+import json
+
+from rest_framework.test import APIClient
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from hs_core.hydroshare.utils import get_resource_types
+
+
+class TestResourceTypes(APITestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_resource_typelist(self):
+        resource_types = set([t.__name__ for t in get_resource_types()])
+
+        response = self.client.get('/hsapi/resourceTypes/', format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        content = json.loads(response.content)
+        rest_resource_types = set([t['resource_type'] for t in content])
+
+        self.assertEqual(resource_types, rest_resource_types)
```

Comment:
Do we document the API publicly yet? This test brings up some thoughts about documentation I'd like to share quickly.

First of all, we *should* document the API. And doing so includes documenting the supported resource types. Second, as the test stands now its basically just replicating the code that the API endpoint uses to fetch the resource types. If something went wrong, and say a resource type stopped being present, this test would still pass because such a failure would likely break both the implementation *and* the test.

Instead, maybe it would be more appropriate for this test to assert the documented resource types are in fact present?